ws-server/src/server.rs

```diff
@@ -427,252 +468,79 @@ async fn background_task(
 
 		tracing::debug!("recv {} bytes", data.len());
 
-		let request_start = middleware.on_request();
+		let request_start = middleware.on_request(remote_addr, &headers);
```

Comment:
So, it looks like those headers will only ever contain Host and Origin from above; what about any custom headers?

Reply:
https://github.com/paritytech/soketto/pull/60,

it's missing from soketto but I don't want to make this PR any larger.